openedx/core/djangoapps/xblock/runtime/blockstore_runtime.py

```diff
@@ -46,6 +49,8 @@ def get_block(self, usage_id, for_parent=None):
             raise TypeError("This runtime can only load blocks stored in Blockstore bundles.")
         try:
             block_type = self.id_reader.get_block_type(def_id)
+            if block_type_overrides and block_type in block_type_overrides:
+                block_type = block_type_overrides[block_type]
         except NoSuchDefinition:
             raise NoSuchUsage(repr(usage_id))  # lint-amnesty, pylint: disable=raise-missing-from
         keys = ScopeIds(self.user_id, block_type, def_id, usage_id)
```

Comment:
Curious how the `block_type` in `ScopeIds` is used? Does it impact the `usage_key`? 🤔

Reply:
My testing indicates that the [`data-usage-key` rendered to the HTML is unchanged (i.e., uses the original block type)](https://github.com/edx/edx-platform/pull/29517/files#diff-11fef21cda98f69e1fb1129e50f0813cd18253bf52adff29f45bc8424446eabeR1015), but the [`data-block-type` uses the overridden block type](https://github.com/edx/edx-platform/pull/29517/files#diff-11fef21cda98f69e1fb1129e50f0813cd18253bf52adff29f45bc8424446eabeR1016).

Is this ok? Are there any other changes to the API response content that you do/don't want to happen?

Reply:
> My testing indicates that the data-usage-key rendered to the HTML is unchanged (i.e., uses the original block type), but the data-block-type uses the overridden block type.

That sounds reasonable.

> Are there any other changes to the API response content that you do/don't want to happen?

Can't think of anything specific.